fitlins/interfaces/afni.py

```diff
@@ -0,0 +1,637 @@
+            #########################
+            # Results are saved to self instead in run_3dREMLfit, if the
+            # memory saving is required it should be implemented there
+            # nistats_flm.labels_.append(labels)
+            # # We save memory if inspecting model details is not necessary
+            # if nistats_flm.minimize_memory:
+            #     for key in results:
+            #         results[key] = SimpleRegressionResults(results[key])
+            # nistats_flm.results_.append(results)
+
+        # Report progress
+        if nistats_flm.verbose > 0:
+            sys.stderr.write(
+                "\nComputation of %d runs done in %i seconds\n\n"
+                % (n_runs, time.time() - t0)
+            )
+
+    def run_3dREMLfit(
+        self, runtime, img_mat, design, contrasts, noise_model, bins, n_jobs
+    ):
+        import nibabel as nb
+        import sys
+
+        tmpdir = op.join(runtime.cwd)
+        self.tmpdir = tmpdir
+        bold_file = op.join(tmpdir, "input.nii.gz")
+        nb.Nifti1Image(img_mat, None).to_filename(bold_file)
+        glt = "glt_results"
+        extra_vars = "glt_extra_variables"
+        t_r = self.inputs.session_info["repetition_time"]
+        stim_labels = self.get_stim_labels()
+        zscores = "zscore_maps"
+        pvals = "pval_maps"
+
+        # Write AFNI style design matrix to file
+        afni_design = get_afni_design_matrix(design, contrasts, stim_labels, t_r)
+        design_fname = op.join(tmpdir, "design.xmat.1D")
+        Path(design_fname).write_text(afni_design)
+
+        # use mask if available:
+        if self.inputs.mask_file:
+            mask_param = f"-mask {self.inputs.mask_file}"
+        else:
+            mask_param = ""
+
+        # Define 3dREMLfit command
+        reml_cmd = f"""\
+        3dREMLfit
+            -matrix {design_fname}
+            -input {bold_file}
+            -Rbuck {glt}
+            -Rvar {extra_vars}
+            -tout
+            -fout
+            -verb
+            {mask_param}
+        """
+
+        # Reformat command string
+        reml_cmd = " ".join(reml_cmd.split())
+
+        # Define commands to get pvals and zscores
+        pval_cmd = f"3dPval -prefix {pvals} {glt}+orig"
+        z_cmd = f"3dPval -prefix {zscores} -zscore {glt}+orig"
+
+        # Execute commands
+        sys.stderr.write(
+            f"3dREMLfit and 3dPval computation will be performed in: {tmpdir}\n"
+        )
+        sp.check_output(reml_cmd, shell=True, cwd=tmpdir, stderr=sp.STDOUT)
+        sp.check_output(pval_cmd, shell=True, cwd=tmpdir, stderr=sp.STDOUT)
+        sp.check_output(z_cmd, shell=True, cwd=tmpdir, stderr=sp.STDOUT)
+
+        pval_bucket, pval_labels = load_bucket_by_prefix(tmpdir, pvals)
+        zscore_bucket, z_labels = load_bucket_by_prefix(tmpdir, zscores)
+        extra_bucket, extra_labels = load_bucket_by_prefix(tmpdir, extra_vars)
+        stat_bucket, stat_labels = load_bucket_by_prefix(tmpdir, glt)
```

Comment:
L332-L361: Is this stuff not already done in [Remlfit](https://nipype.readthedocs.io/en/latest/interfaces/generated/interfaces.afni/model.html#remlfit)? I assumed all of the parsing work should already be done. If it's not, it would make more sense to fix up that interface than to have a separate implementation here.

I guess we need to add a 3dPval interface. We can build that here or in nipype. (It'll get upstreamed eventually.)

Reply:
Sounds good. That does seem like 30 lines of superfluous code. I didn't want to complicate things before a refactor. Would you have just defined and executed the single interface node within the class method? 

I can take a hack at writing the 3dPval interface

Reply:
My thought here was to just use the interface, not wrapped in a `Node()`, which means it would just use the working directory of the `Node` that the whole interface is being run in.

Reply:
Great, will do.